proposals/semi-auto-properties.md

```diff
@@ -10,6 +10,257 @@ Two common scenarios are that you want to apply a constraint on the setter, ensu
 
 In these cases by now you always have to create an instance field and write the whole property yourself.  This not only adds a fair amount of code, but it also leaks the `field` into the rest of the type's scope, when it is often desirable to only have it be available to the bodies of the accessors.
 
+## Detailed design
+
+For properties with an `init` accessor, everything that applies below to `set` would apply instead to the `init` accessor.
+
+**Principle 1:** Every property declaration can be thought of as having a backing field by default, which is elided when not used. The field is referenced using the keyword `field` and its visibility is scoped to the accessor bodies.
+
+**Principle 2:** `get;` will now be considered syntactic sugar for `get => field;`, and `set;` will now be considered syntactic sugar for `set => field = value;`.
+
+Both of these principles only apply under the same conditions where `{ get; }` or `{ get; set; }` already declares an auto property ([§15.7.4](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/classes.md#1574-automatically-implemented-properties)). Indexers also remain unaffected.
+
+This means that properties may now mix and match auto accessors with full accessors. For example:
```

Comment:
"auto accessor" is an interesting name.  Is it somethign we already use elsewhere?  It might be worth defining it above to ensure it's clear what that is referring to.

Reply:
similarly, what a "full accessor" is.   

Something as simple as "we call accessors without a body  (e.g. `get;` and `set;`) an "auto accessor".  We call accessors with a body (e.g. `get { ... }` and `get => ...`) a "full accessor".

Reply:
The spec introduces "auto-property" as short for "automatically-implemented property," so what I would reach for to address this is introducing "auto-accessor" as short for "automatically-implemented accessor."

Reply:
i think that's fine.  I think we should start though with a glossery of terms we want to use here and what they mean.  :)